cloud-watch-to-slack-testing-58006136-c30b-43d1-880c-7b36f860e7c8/deployment/index.js

```diff
@@ -142,10 +146,40 @@ function sendMessageToSlack(messageText, callback) {
   });
 }
 
+// Set the Slack channel based on the input SNS Topic to Slack Channel map
+// in the snsTopicToSlackChannel env var with format
+// {"<SNS Topic resource id>":"<slack channel name>"}
+// E.g.
+//     {"slack-low-priority-topic":"dockstore-testing",
+//     "slack-medium-priority-topic":"dockstore-dev-alerts",
+//     "slack-high-priority-topic":"dockstore-alerts",
+//     "SSM-to-Slack-SNSTopicToSlack-1VI4KZW1DSADS":"dockstore-dev-testing" }
+// input: SNS topic ARN
+function setSlackChannelBasedOnSNSTopic(topicArn) {
+  // Get the SNS Topic resource ID from the AWS ARN
+  // https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
+  const snsTopicResourceID = topicArn.slice(topicArn.lastIndexOf(":") + 1);
+  const snsTopicToSlackChannel = JSON.parse(`${snsTopicToSlackChannelJSON}`);
```

Comment:
This is an environment configuration, right? If so, it would be better if you parsed it once, probably on line 29. This lambda will get so little traffic that it probably doesn't matter that much, so not crucial.

Reply:
fixed